commerce_coordinator/apps/titan/tasks.py

```diff
@@ -102,8 +101,22 @@ def payment_processed_save_task(
             payment_state_paid_cache_key = get_paid_payment_state_cache_key(payment_number)
             TieredCache.set_all_tiers(payment_state_paid_cache_key, payment, settings.DEFAULT_TIMEOUT)
         elif payment_state == PaymentState.FAILED.value:
+            if 'metadata' in provider_response_body:
+                if 'payment_number' in provider_response_body['metadata']:
+                    provider_response_body['metadata']['payment_number'] = ''
+                if 'order_number' in provider_response_body['metadata']:
+                    order_uuid = provider_response_body['metadata']['order_number']
+            titan_api_client.create_payment(
+                order_uuid=order_uuid,
+                reference_number=reference_number,
+                payment_method_name=PaymentMethod.STRIPE.value,
+                provider_response_body=provider_response_body,
```

Comment:
This `provider_response_body` is Stripe event with failed payment information. We are already saving this information to Titan system with our failed payment (l[ine 96](https://github.com/edx/commerce-coordinator/pull/107/files#diff-bf55193548ab769a028ca42527203ecd61a500970277a1881d13390589e12a54L97))

In the original follow, We use the payload that we get from Stripe while creating payment intent. As we don't have that here, using `provider_response_body=None` would make more sense.

@pshiu what would you say?

Reply:
or maybe we can use `StripeApiClient.retrieve_payment_intent` to get provide_response_body.

Reply:
changed to retrieve call since provider_response_body is a required field